.github/workflows/nightly-new-viewer.yml

```diff
@@ -0,0 +1,57 @@
+name: Nightly builds
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+    
+
+jobs:
+  build-stable:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16.x"
+
+      - name: Cache dependencies
```

Comment:
can you explain what this does for us?  this might be a good devchat topic to understand how it works

Reply:
I'm not sure if I was the one who originally added this, but here is the github article about it. It just speeds up the run time as I understand it. 

https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows